src/__tests__/scheduling/appointments/new/NewAppointment.test.tsx

```diff
@@ -97,212 +103,150 @@ describe('New Appointment', () => {
-
-      await act(async () => {
-        await onClick()
+      fireEvent.change(container.querySelectorAll('.react-datepicker__input-container input')[1], {
+        target: { value: expectedAppointment.endDateTime },
       })
-      wrapper.update()
-      const alert = wrapper.find(Alert)
-      const typeahead = wrapper.find(Typeahead)
-      const dateInput = wrapper.find(DateTimePickerWithLabelFormGroup).at(0)
+      userEvent.click(screen.getByText(/actions\.save/i))
 
+      expect(screen.getByText(expectedError.message)).toBeInTheDocument()
+      expect(screen.getByPlaceholderText(/scheduling\.appointment\.patient/i)).toHaveClass(
+        'is-invalid',
+      )
+      expect(container.querySelectorAll('.react-datepicker__input-container input')[0]).toHaveClass(
+        'is-invalid',
+      )
+      expect(screen.getByText(expectedError.startDateTime)).toBeInTheDocument()
       expect(AppointmentRepository.save).toHaveBeenCalledTimes(0)
-      expect(alert.prop('message')).toEqual(expectedError.message)
-      expect(typeahead.prop('isInvalid')).toBeTruthy()
-      expect(dateInput.prop('isInvalid')).toBeTruthy()
-      expect(dateInput.prop('feedback')).toEqual(expectedError.startDateTime)
     })
 
     it('should call AppointmentRepo.save when save button is clicked', async () => {
-      let wrapper: any
-      await act(async () => {
-        wrapper = await setup()
-      })
+      const { container } = setup()
 
       const expectedAppointment = {
-        patient: '123',
+        patient: testPatient.fullName,
         startDateTime: roundToNearestMinutes(new Date(), { nearestTo: 15 }).toISOString(),
         endDateTime: addMinutes(
           roundToNearestMinutes(new Date(), { nearestTo: 15 }),
           60,
         ).toISOString(),
         location: 'location',
         reason: 'reason',
-        type: 'type',
+        type: 'routine',
       } as Appointment
 
-      act(() => {
-        const appointmentDetailForm = wrapper.find(AppointmentDetailForm)
-        const onFieldChange = appointmentDetailForm.prop('onFieldChange')
-        onFieldChange('patient', expectedAppointment.patient)
-      })
-
-      wrapper.update()
-
-      act(() => {
-        const appointmentDetailForm = wrapper.find(AppointmentDetailForm)
-        const onFieldChange = appointmentDetailForm.prop('onFieldChange')
-        onFieldChange('startDateTime', expectedAppointment.startDateTime)
-      })
-
-      wrapper.update()
-
-      act(() => {
-        const appointmentDetailForm = wrapper.find(AppointmentDetailForm)
-        const onFieldChange = appointmentDetailForm.prop('onFieldChange')
-        onFieldChange('endDateTime', expectedAppointment.endDateTime)
-      })
-
-      wrapper.update()
+      userEvent.type(
+        screen.getByPlaceholderText(/scheduling\.appointment\.patient/i),
+        expectedAppointment.patient,
+      )
+      userEvent.click(await screen.findByText(`${testPatient.fullName} (${testPatient.code})`))
```

Comment:
Didn't know you could use await here!! Obvious now I think about it you need to wait for the type event to complete before you click save.

Reply:
@emma-r-slight Almost! What happens with the Typeahead component - at least for when looking up a patient - is the typing input is debounced while it searches the database for Patients that match. So without `await screen.findBy*`, it just fails because there aren't any results yet. Whereas the "Type" of the appointment is a fixed list, so it can just directly works... (even though the test code treats it the same way).